instance/models/openedx_instance.py

```diff
@@ -449,24 +451,27 @@ def require_user_creation_success(self):
     def terminate_obsolete_appservers(self, days=2):
         """
         Terminate app servers that were created more than `days` before now, except:
-        - the active appserver if there is one,
+        - the active appserver(s) if there are any,
         - a release candidate (rc) appserver, to allow testing before the next appserver activation
           (we keep the most recent running appserver)
         - a fallback appserver, for `days` after activating an appserver, to allow reverts
-          (we keep the most recent running appserver created before the current activation)
+          (we keep the most recent running appserver created before the latest activation)
         """
-        active_appserver = self.active_appserver
+        latest_active_appserver = None
+        if self.get_active_appservers().count():
+            latest_active_appserver = self.get_active_appservers().latest('last_activated')
         fallback_appserver = None
         rc_appserver = None
         now = timezone.now()
 
         for appserver in self.appserver_set.all().order_by('-created'):
-            if appserver == active_appserver:
+            # Skip active appservers
+            if appserver.is_active:
                 continue
 
-            # Keep a running appserver as fallback for `days` after activation, to allow reverts
-            if active_appserver and appserver.created < active_appserver.last_activated:
-                if not sufficient_time_passed(active_appserver.last_activated, now, days) \
+            # Keep a running appserver as fallback for `days` after latest activation, to allow reverts
```

Comment:
> More complex but better is to introduce an AppServerSet as another layer - likely out of scope for this though?

Yes, this seem like the right approach to me.

> We could change the UI to have a "Spawn ___ AppServers" button, and every time you use that, it spawns N appservers and groups them together with an auto-generated "generation ID" or something? Or the "N" here could be a property of the instance, which would be simpler.

It has the advantage of still grouping the servers logically and might be smaller. But the gain in implementation time now might be repaid by making the logic to handle groups of servers more complex, since it wouldn't be encapsulated by a dedicated class. So I'm not a huge fan, but given the urgency context, that might still be a reasonable middle ground - but imho we would need a proper follow-up story to clean it up and implement a proper AppServerSet class.

> Or we could have a numeric setting on each instance indicating how many active servers are expected to be running at any given time. Then we could use that as the maximum number of fallback/release candidate servers to be kept running.

I'd rather have an explicit grouping of servers. This would likely make the logic even more complex, and increase the odds of subtle issues - for example if multiple server creations are requested in a short amount of time, or when the number of servers on the instance changes.